src/controllers/controller.line.js

```diff
@@ -209,6 +209,19 @@ module.exports = function(Chart) {
 			};
 		},
 
+		calculatePointX: function(value, index, datasetIndex){
+			var me = this;
+			var meta = me.getMeta();
+			var labels = me.chart.data.labels;
+			var dataset = me.getDataset();
+			var xScale = me.getScaleForId(meta.xAxisID);
+			if (labels && labels.length === 1 || dataset.data.length === 1) {
+				return xScale.getPixelForValue(typeof value === 'object' ? value : NaN, index, datasetIndex, true);
+			} else {
+				return xScale.getPixelForValue(typeof value === 'object' ? value : NaN, index, datasetIndex, me.chart.isCombo);
+			}
+		},
```

Comment:
I also don't like extra getMeta(), getDataset() etc in calculatePointX(). It was added "for symmetry" bacause of calculatePointY() exists, but it doesn't contain any complex logic unlike calculatePointY(), so I like your short solution more.